### PR TITLE
Do not collect and skip non-disabled tests when rerunning disabled tests

### PR DESCRIPTION
**Pull request by huydhn**
The console log blows up to much when running in rerun disabled tests mode (x50) https://hud.pytorch.org/pytorch/pytorch/commit/e132f09e8878418fb98a4b76a441a324452354ec.  Each log is around 1GB and the whole uncompressed logs is ~50GB.  After compression, it will be around 1GB, still too big.  The increase comes mainly from the multiple SKIPPED message for non-disabled tests, which is expected due to how SkipTest and pytest-flakyfinder currently work.

I update `test/conftest.py` to completely ignore skipped tests when rerunning disabled test instead of collecting then skipping 50 tests each.  The benefit of doing is is much more than I originally expect:
  * Rerun disabled tests jobs now finish in less than half an hour as they should be
  * Fix OOM runner crash because of too many collected tests
  * Fix verbosity issue as now only disabled tests are run x50 times.  There are only few hundreds of them atm
  * Fix timed out issue when rerunning disabled distributed and ASAN tests.  They are just too slow when running at x50

### Testing

When rerunning disabled tests https://github.com/pytorch/pytorch/actions/runs/5084508614, only disabled tests on the platform are run, for example `test_ops_jit` on https://ossci-raw-job-status.s3.amazonaws.com/log/13770164954 only ran 100 tests (`test_variant_consistency_jit_linalg_lu_cuda_float32` + `test_variant_consistency_jit_linalg_lu_factor_cuda_complex64`) x50.

```
Executing ['/opt/conda/envs/py_3.10/bin/python', '-bb', 'test_ops_jit.py', '--shard-id=1', '--num-shards=2', '-v', '-vv', '-rfEX', '-p', 'no:xdist', '--use-pytest', '--sc=test_ops_jit_1', '--flake-finder', '--flake-runs=50', '--import-slow-tests', '--import-disabled-tests', '--rerun-disabled-tests'] ... [2023-05-25 21:32:49.763856]

Expand the folded group to see the log file of test_ops_jit 2/2
##[group]PRINTING LOG FILE of test_ops_jit 2/2 (/var/lib/jenkins/workspace/test/test-reports/test_ops_jit_h2wr_t2c.log)
Test results will be stored in test-reports/python-pytest/test_ops_jit/test_ops_jit-51a83bd44549074e.xml
============================= test session starts ==============================
platform linux -- Python 3.10.11, pytest-7.3.1, pluggy-1.0.0 -- /opt/conda/envs/py_3.10/bin/python
cachedir: .pytest_cache
hypothesis profile 'pytorch_ci' -> database=None, max_examples=50, derandomize=True, suppress_health_check=[HealthCheck.too_slow]
rootdir: /var/lib/jenkins/workspace
configfile: pytest.ini
plugins: hypothesis-5.35.1, cpp-2.3.0, flakefinder-1.1.0, rerunfailures-11.1.2, shard-0.1.2, xdist-3.3.0, xdoctest-1.1.0
collecting ... collected 1084 items
Running 100 items in this shard: test/test_ops_jit.py::TestJitCUDA::test_variant_consistency_jit_linalg_lu_cuda_float32 (x50), test/test_ops_jit.py::TestJitCUDA::test_variant_consistency_jit_linalg_lu_factor_cuda_complex64 (x50)
stepcurrent: Cannot find last run test, not skipping

test_ops_jit.py::TestJitCUDA::test_variant_consistency_jit_linalg_lu_cuda_float32 PASSED [2.1876s] [  1%]
test_ops_jit.py::TestJitCUDA::test_variant_consistency_jit_linalg_lu_factor_cuda_complex64 PASSED [4.5615s] [  2%]
```

* [pull](https://github.com/pytorch/pytorch/actions/runs/5093566864)
* [trunk](https://github.com/pytorch/pytorch/actions/runs/5095364311)
* [periodic](https://github.com/pytorch/pytorch/actions/runs/5095378850)
* [slow](https://github.com/pytorch/pytorch/actions/runs/5095390285)

### CODE REVIEW

**Review by clee2000 — APPROVED** [review]
dont forget to change filter_test_configs back before merging

File: .github/scripts/filter_test_configs.py [hunk 1]
```diff
@@ -386,10 +386,11 @@ def main() -> None:
         # No PR number, no tag, we can just return the test matrix as it is
         filtered_test_matrix = test_matrix
 
-    if args.event_name == "schedule" and args.schedule == "29 8 * * *":
-        # we don't want to run the mem leack check or disabled tests on normal
-        # periodically scheduled jobs, only the ones at this time
-        filtered_test_matrix = set_periodic_modes(filtered_test_matrix, args.job_name)
+    # DEBUG: TO BE REVERTED
+    # if args.event_name == "schedule" and args.schedule == "29 8 * * *":
+    # we don't want to run the mem leack check or disabled tests on normal
```
Comment (malfet) [inline .github/scripts/filter_test_configs.py:391]:
```suggestion
    # we don't want to run the mem leak check or disabled tests on normal
```

File: test/conftest.py [hunk 2]
```diff
@@ -214,6 +219,53 @@ def pytest_report_teststatus(report, config):
         )
 
 
+@pytest.hookimpl(trylast=True)
+def pytest_collection_modifyitems(items: List[Any]):
```
Comment (malfet) [inline test/conftest.py:223]:
```suggestion
def pytest_collection_modifyitems(items: List[Any]) -> None:
```

File: test/conftest.py [hunk 3]
```diff
@@ -214,6 +219,53 @@ def pytest_report_teststatus(report, config):
...
+            for disabled_test in json.load(fp):
+                m = disabled_regex.match(disabled_test)
+                if m:
+                    test_name = m["test_name"]
+                    test_class = m["test_class"]
+                    disabled_tests[test_class].add(test_name)
+    else:
+        return
+
+    # When rerunning disabled test, ignore all test cases that are not disabled
```
Comment (malfet) [inline test/conftest.py:249]:
```suggestion
    if not disabled_tests_file and os.path.exists(disabled_tests_file):
        return
    with open(disabled_tests_file) as fp:
        for disabled_test in json.load(fp):
            m = disabled_regex.match(disabled_test)
            if m:
                test_name = m["test_name"]
                test_class = m["test_class"]
                disabled_tests[test_class].add(test_name)

    # When rerunning disabled test, ignore all test cases that are not disabled
```

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**